lambda/src/firebase-authorizer/firebase-authorizer.js

```diff
@@ -73,79 +73,74 @@ function checkUserAuthorization(event, context, callback) {
+        uid: decodedToken.uid,
+        email: decodedToken.email || null,
+        email_verified: decodedToken.email_verified || false,
+        auth_time: decodedToken.auth_time || 0,
       }
 
-      // Validate the Firebase token.
-      admin
-        .auth()
-        .verifyIdToken(token)
-        .then(decodedToken => {
-          const user = {
-            uid: decodedToken.uid,
-            email: decodedToken.email || null,
-            email_verified: decodedToken.email_verified || false,
-            auth_time: decodedToken.auth_time || 0,
-          }
+      // Conditions for authorization. We do not check for a valid
+      // email because we create the user before email validation.
+      const valid = !!user.uid
 
-          // Conditions for authorization. We do not check for a valid
-          // email because we create the user before email validation.
-          const valid = !!user.uid
-
-          // Generate AWS authorization policy
-          callback(null, generatePolicy(user, valid, event.methodArn))
-        })
-        .catch(e => {
-          console.error(e)
-          callback('Error: Invalid token')
-        })
+      // Generate AWS authorization policy
+      return generatePolicy(user, valid, event.methodArn)
     } catch (e) {
       console.error(e)
-      callback('Error: Invalid token')
+      throw new Error('Error: Invalid token')
     }
+  } catch (e) {
+    console.error(e)
+    throw new Error('Error: Invalid token')
   }
 }
 
-const handler = (event, context, callback) => {
+const handler = async event => {
   // Decrypt secure environment variables.
   if (decryptedFirebasePrivateKey) {
-    checkUserAuthorization(event, context, callback)
-  } else {
-    // Decrypt code should run once and variables stored outside of the function
-    // handler so that these are decrypted once per container
-    const kms = new AWS.KMS()
-    kms.decrypt(
-      { CiphertextBlob: Buffer.from(encryptedFirebasePrivateKey, 'base64') },
-      (err, data) => {
-        if (err) {
-          console.log('Decrypt error:', err)
-          callback('Error decrypting secure environnment variables.')
-          return
-        }
-        decryptedFirebasePrivateKey = data.Plaintext.toString('ascii')
-        checkUserAuthorization(event, context, callback)
-      }
-    )
+    return checkUserAuthorization(event)
   }
+  // Decrypt code should run once and variables stored outside of the function
+  // handler so that these are decrypted once per container
+  const kms = new AWS.KMS()
+  kms.decrypt(
+    { CiphertextBlob: Buffer.from(encryptedFirebasePrivateKey, 'base64') },
+    (err, data) => {
+      if (err) {
+        console.log('Decrypt error:', err)
+        return 'Error decrypting secure environnment variables.'
+      }
+      decryptedFirebasePrivateKey = data.Plaintext.toString('ascii')
+      return checkUserAuthorization(event)
+    }
+  )
+
+  return new Error('Error: Unexpected end of function')
```

Comment:
Because the KMS stuff above is callback-based, I think you'll be reaching this return any time the decrypted private key is not available.